packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,702 @@
+  // ...
+
+  // score audit based on budget
+  const errorCount = data.filter(({ size }) => options.budget < size).length;
+  fileSizeAuditOutput: AuditOutput = {
+    ...fileSizeAuditOutput,
+    // score is factor of over-budget files to all files
+    score: factorOf(issues, errorCount),
+    value: errorCount,
+    displayValue: pluralizeToken('file', errorCount),
+  };
+
+  return [fileSizeAuditOutput];
+}
+```
+
+### Audit groups
+
+As an optional property a plugin can maintain `groups` as [AuditGroup](@TODO).
+Where [categories](@TODO) can score audits across plugins groups are only targeting plugins within a plugin.
+For simple plugins this is not needed, but can benefit as audit groups also simplify the configuration.
+
+A audit group maintains:
+
+- metadata about the group [AuditGroupMeta](@TODO)
+- a list of referenced audits under `refs` as [AuditGroupRef](@TODO) array
+
+The minimum information of an audit group looks like this:
+
+```typescript
+import { AuditGroup } from '@code-pushup/models';
+
+const myAuditGroup: AuditGroup = {
+  slug: 'performance',
+  refs: [
+    {
+      slug: 'file-size-audit',
+      weight: 1,
+    },
+  ],
+  // optional details
+};
+```
+
+The weight property of a reference is used to calculate a score for all audits listed under refs.  
+A weight of 0 can be used for informative audits and a value from 1 upward to give a weight in the score compared to other audits.
+
+### Audit details
+
+To have better attribution in your audits you can use the `details` section in `AuditOutputs`.
+This helps to make the plugin results more actionable and valuable for the user.
+
+We will extend the `fileSizeAuditOutput` with `details` show which files exceed the defined budget.
+
+**file-size plugin runnerFunction**
+
+```typescript
+// file-size.plugin.ts
+// ...
+import { basename } from 'path';
+import { formatBytes, toUnixPath } from '@code-pushup/utils';
+import { AuditOutput } from './plugin-process-output';
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  // ...
+
+  // assert file size information with budget
+  const issues = data.map(({ file, size }) => assertFileSizeInfo(file, size, options.budget));
+
+  // add details if issues given
+  if (issues.length) {
+    fileSizeAuditOutput = {
+      ...fileSizeAuditOutput,
+      details: {
+        issues,
+      },
+    };
+  }
+
+  return [fileSizeAuditOutput];
+}
+
+// assert file size info with budget
+export function assertFileSize(file: string, size: number, budget?: number): Issue {
+  const auditOutputBase = {
+    source: {
+      // format path to be in scope of the repository
+      file: toUnixPath(file, { toRelative: true }),
+    },
+  } satisfies AuditOutput['source'];
+
+  if (budget && budget < size) {
+    return {
+      ...auditOutputBase,
+      severity: 'error',
+      message: `File ${basename(filePath)} is ${formatBytes(size - budget)} bytes too big. ( budget: ${formatBytes(budget)})`,
```

Comment:
Since `formatBytes` already returns a suitable suffix, this would produce `26.7 kB bytes too big` instead of `26.7 kB too big`.

```suggestion
      message: `File ${basename(filePath)} is ${formatBytes(size - budget)} too big. ( budget: ${formatBytes(budget)})`,
```